src/Entities/Entity.cpp

```diff
@@ -427,7 +427,7 @@ bool cEntity::DoTakeDamage(TakeDamageInfo & a_TDI)
 
 		if (SharpnessLevel > 0)
 		{
-			a_TDI.FinalDamage += static_cast<int>(ceil(1.25 * SharpnessLevel));
+			a_TDI.FinalDamage += static_cast<float>(ceil(1.25 * SharpnessLevel));
```

Comment:
Just get rid of the entire cast, and `ceil`:
`a_TDI.FinalDamage += 1.25f * SharpnessLevel;`
(This applies to other changes of yours as well)